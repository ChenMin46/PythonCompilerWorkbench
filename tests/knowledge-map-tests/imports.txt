import foo
from  foo  import bar
from   foo    import bar   ,  baz
from foo import  *
import foo.bar
from foo.bar import baz

# semicolons to start new lines
import foo.bar; x += 5; y -= 5
from foo.bar import baz; baz()
