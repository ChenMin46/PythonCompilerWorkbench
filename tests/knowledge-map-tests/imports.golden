import foo
from  foo  import bar
from   foo    import bar   ,  baz
from foo import  *
import foo.bar
from foo.bar import baz

# semicolons to start new lines
import foo.bar; x += 5; y -= 5
from foo.bar import baz; baz()
===
Module
  body: [
    Import >> 'import foo'
      names: [
        alias
          name: 'foo'
          asname: None
      ]
    ImportFrom >> 'from  foo  import bar'
      module: 'foo'
      names: [
        alias
          name: 'bar'
          asname: None
      ]
      level: 0
    ImportFrom >> 'from   foo    import bar   ,  baz'
      module: 'foo'
      names: [
        alias
          name: 'bar'
          asname: None
        alias
          name: 'baz'
          asname: None
      ]
      level: 0
    ImportFrom >> 'from foo import  *'
      module: 'foo'
      names: [
        alias
          name: '*'
          asname: None
      ]
      level: 0
    Import >> 'import foo.bar'
      names: [
        alias
          name: 'foo.bar'
          asname: None
      ]
    ImportFrom >> 'from foo.bar import baz'
      module: 'foo.bar'
      names: [
        alias
          name: 'baz'
          asname: None
      ]
      level: 0
    Import >> 'import foo.bar'
      names: [
        alias
          name: 'foo.bar'
          asname: None
      ]
    AugAssign >> 'x += 5'
      target: Name >> 'x'
        id: 'x'
        ctx: Store
      op: Add
      value: Num >> '5'
        n: 5
    AugAssign >> 'y -= 5'
      target: Name >> 'y'
        id: 'y'
        ctx: Store
      op: Sub
      value: Num >> '5'
        n: 5
    ImportFrom >> 'from foo.bar import baz'
      module: 'foo.bar'
      names: [
        alias
          name: 'baz'
          asname: None
      ]
      level: 0
    Expr
      value: Call >> 'baz()'
        func: Name >> 'baz'
          id: 'baz'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            "import foo"
          ], 
          "id": "id_0", 
          "name": "Import"
        }, 
        "\n", 
        {
          "contents": [
            "from  foo  import bar"
          ], 
          "id": "id_0", 
          "name": "ImportFrom"
        }, 
        "\n", 
        {
          "contents": [
            "from   foo    import bar   ,  baz"
          ], 
          "id": "id_0", 
          "name": "ImportFrom"
        }, 
        "\n", 
        {
          "contents": [
            "from foo import  *"
          ], 
          "id": "id_0", 
          "name": "ImportFrom"
        }, 
        "\n", 
        {
          "contents": [
            "import foo.bar"
          ], 
          "id": "id_0", 
          "name": "Import"
        }, 
        "\n", 
        {
          "contents": [
            "from foo.bar import baz"
          ], 
          "id": "id_0", 
          "name": "ImportFrom"
        }, 
        "", 
        "\n", 
        "", 
        "\n", 
        "", 
        {
          "id": "tid_1", 
          "name": "comment", 
          "value": "# semicolons to start new lines"
        }, 
        "\n", 
        "", 
        {
          "contents": [
            "import foo.bar"
          ], 
          "id": "id_0", 
          "name": "Import"
        }, 
        {
          "id": "tid_2", 
          "name": "token", 
          "value": ";"
        }, 
        " ", 
        {
          "contents": [
            {
              "contents": [
                "x"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_3", 
              "name": "token", 
              "value": "+="
            }, 
            " ", 
            {
              "contents": [
                "5"
              ], 
              "id": "id_0", 
              "name": "Num"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }, 
        {
          "id": "tid_4", 
          "name": "token", 
          "value": ";"
        }, 
        " ", 
        {
          "contents": [
            {
              "contents": [
                "y"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_5", 
              "name": "token", 
              "value": "-="
            }, 
            " ", 
            {
              "contents": [
                "5"
              ], 
              "id": "id_0", 
              "name": "Num"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }, 
        "\n", 
        {
          "contents": [
            "from foo.bar import baz"
          ], 
          "id": "id_0", 
          "name": "ImportFrom"
        }, 
        {
          "id": "tid_6", 
          "name": "token", 
          "value": ";"
        }, 
        " ", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "baz"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                "()"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
